src/tls_stream.rs

```diff
@@ -730,18 +730,21 @@ impl<S> TlsStream<S>
     fn encrypt(&mut self, buf: &[u8], sizes: &winapi::SecPkgContext_StreamSizes) -> io::Result<()> {
         assert!(buf.len() <= sizes.cbMaximumMessage as usize);
 
-        unsafe {
-            let len = sizes.cbHeader as usize + buf.len() + sizes.cbTrailer as usize;
+        let len = sizes.cbHeader as usize + buf.len() + sizes.cbTrailer as usize;
 
-            if self.out_buf.get_ref().len() < len {
-                self.out_buf.get_mut().resize(len, 0);
-            }
+        self.out_buf.set_position(0);
+        if self.out_buf.get_ref().len() < len {
```

Comment:
This is redundant, resize [implies](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.resize) truncate for `len() < len`.